src/hooks/app/useUpdateStatus.ts

```diff
@@ -0,0 +1,90 @@
+import * as Sentry from '@sentry/react';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { check, Update } from '@tauri-apps/plugin-updater';
+import { relaunch } from '@tauri-apps/plugin-process';
+
+import { useAppStateStore } from '@app/store/appStateStore';
+import { useAppConfigStore } from '@app/store/useAppConfigStore';
+import { useUIStore } from '@app/store/useUIStore';
+import { useInterval } from '../helpers/useInterval';
+
+const UPDATE_CHECK_INTERVAL = 1000 * 60 * 60; // 1 hour
+
+export const useHandleUpdate = () => {
```

Comment:
Please verify that download progress works well. I had issue with tauri js API where it would get stuck from massive amounts of incoming `Progress` events. In the end rust API worked much better with high amount of events.

Reply:
we did check this with a draft auto update release a long while back and it seemed alright, i didn't get stuck :D but that hasn't been tested recently